astropy/modeling/parameters.py

```diff
@@ -728,22 +728,22 @@ def __rtruediv__(self, val):
         return val / self.value
 
     def __eq__(self, val):
-        return (np.asarray(self) == np.asarray(val)).all()
+        return np.asanyarray(self) == np.asanyarray(val)
```

Comment:
I think you are right that is fine in this case; in principle, the `self.__array__().__eq__(val)` could return `NotImplemented`, which would then mean `val` would have a chance to try (which might recognize `Parameter` and want to do something else than compare with `value`. But this is probably too far-fetched!